### PR TITLE
Make PairClassification consistent with other tasks

### LINKED ISSUE #1037: Standardize scores structure for tasks
I've created a PR https://github.com/embeddings-benchmark/mteb/pull/1034 to address an issue where `main_score` was not consistently appearing in the scores "at top level", particularly affecting `PairClassification` and `Reranking` tasks (e.g., see the [TERRa results](https://github.com/embeddings-benchmark/mteb/blob/main/results/sergeyzh__rubert-tiny-turbo/8ce0cf757446ce9bb2d5f5a4ac8103c7a1049054/TERRa.json)). I propose changing the structure from:

```json
"cosine": {
  "accuracy": 0.5960912052117264,
  "ap": 0.5608760299515377
  // other metrics
}
```

to:

```json
"cosine_accuracy": 0.5960912052117264,
"cosine_ap": 0.5608760299515377
// other metrics similarly structured
```

This approach aligns with what was previously done with STS metrics and may provide clearer and more consistent results across different tasks.

----

edit by @KennethEnevoldsen: 
This would notably lead to breaking changes in the "ap" (which is "max_ap") scores for PairClassification and similarly for reranking. A solution for backward compatibility is to set "max_ap" -> "ap".

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist
- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

Closes #1037.
I also run tests similar to #1034

### CODE REVIEW

File: mteb/tasks/PairClassification/deu/FalseFriendsDeEnPC.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class FalseFriendsDeEnPC(AbsTaskPairClassification):
         category="s2s",
         eval_splits=["test"],
         eval_langs=["deu-Latn"],
-        main_score="ap",
+        main_score="cosine_ap",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/PairClassification/deu/FalseFriendsDeEnPC.py:21]: wasn't it max_ap before?

File: mteb/tasks/PairClassification/eng/LegalBenchPC.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ class LegalBenchPC(AbsTaskPairClassification):
         category="s2s",
         eval_splits=["test"],
         eval_langs=["eng-Latn"],
-        main_score="accuracy",
+        main_score="cosine_accuracy",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/PairClassification/eng/LegalBenchPC.py:78]: wasn't it max_accuracy before?

File: mteb/abstasks/AbsTaskPairClassification.py [hunk 3]
```diff
@@ -51,16 +50,5 @@ def _evaluate_subset(
...
-        # Compute max
-        max_scores = defaultdict(list)
-        for sim_fct in scores:
-            for metric in ["accuracy", "f1", "ap"]:
-                max_scores[metric].append(scores[sim_fct][metric])
-
-        for metric in max_scores:
-            max_scores[metric] = max(max_scores[metric])
-
-        scores["max"] = dict(max_scores)
```
Comment (KennethEnevoldsen) [inline mteb/abstasks/AbsTaskPairClassification.py:63]: This should still be there

File: mteb/evaluation/evaluators/PairClassificationEvaluator.py [hunk 4]
```diff
@@ -164,11 +166,7 @@ def _compute_metrics(
 
         return {
             "accuracy": float(acc),
-            "accuracy_threshold": float(acc_threshold),
             "f1": float(f1),
-            "f1_threshold": float(f1_threshold),
-            "precision": float(precision),
-            "recall": float(recall),
```
Comment (KennethEnevoldsen) [inline mteb/evaluation/evaluators/PairClassificationEvaluator.py:171]: Would prefer not to loose metrics

**Review by KennethEnevoldsen — APPROVED** [review]
Thanks. Should we add back in the missing scores. Then I also believe the main_score for "ap" corresponds to "max_ap" and not "cosine_ap" though cosine ap is often the max. 

@Muennighoff how to we ensure that "max_ap" and "ap" are both valid scores in the leaderboard?

File: mteb/abstasks/AbsTaskPairClassification.py (hunk 3 above)
Comment (Samoed) [reply mteb/abstasks/AbsTaskPairClassification.py:63]: Moved to PairClassificationEvaluator

File: mteb/evaluation/evaluators/PairClassificationEvaluator.py (hunk 4 above)
Comment (Samoed) [reply mteb/evaluation/evaluators/PairClassificationEvaluator.py:171]: I removed they because previously they weren't added to dataset

**Review by KennethEnevoldsen — COMMENTED** [review]
Thanks for this. The PR is looking very promising. Will you do a run of a model to ensure that the results looks reasonable. Assuming that looks alright we can merge it in.

**Comment by KennethEnevoldsen** [comment]
Ran an example here

```python
{
  "dataset_revision": "7335288588f14e5a687d97fc979194c2abe6f4e7",
  "evaluation_time": 11.970552682876587,
  "kg_co2_emissions": null,
  "mteb_version": "1.12.67",
  "scores": {
    "test": [
      {
        "cosine_accuracy": 0.6239067055393586,
        "cosine_accuracy_threshold": 0.7745091319084167,
        "cosine_ap": 0.6484016684922852,
        "cosine_f1": 0.6789940828402367,
        "cosine_f1_threshold": 0.6167908310890198,
        "cosine_precision": 0.5510204081632653,
        "cosine_recall": 0.884393063583815,
        "dot_accuracy": 0.5519922254616132,
        "dot_accuracy_threshold": 8.397539138793945,
        "dot_ap": 0.5647177517881756,
        "dot_f1": 0.6714100905562743,
        "dot_f1_threshold": 0.8443644642829895,
        "dot_precision": 0.5053554040895814,
        "dot_recall": 1.0,
        "euclidean_accuracy": 0.6190476190476191,
        "euclidean_accuracy_threshold": 2.305285930633545,
        "euclidean_ap": 0.6052430762311987,
        "euclidean_f1": 0.6822498173849526,
        "euclidean_f1_threshold": 3.0024242401123047,
        "euclidean_precision": 0.5494117647058824,
        "euclidean_recall": 0.8998073217726397,
        "hf_subset": "default",
        "languages": [
          "fas-Arab"
        ],
        "main_score": 0.6484016684922852,
        "manhattan_accuracy": 0.6229348882410107,
        "manhattan_accuracy_threshold": 38.67555236816406,
        "manhattan_ap": 0.606826462659015,
        "manhattan_f1": 0.682124158563949,
        "manhattan_f1_threshold": 45.126319885253906,
        "manhattan_precision": 0.5574572127139364,
        "manhattan_recall": 0.8786127167630058,
        "max_ap": 0.6484016684922852,
        "max_f1": 0.6822498173849526,
        "max_precision": 0.5574572127139364,
        "max_recall": 1.0,
        "similarity_accuracy": 0.6239067055393586,
        "similarity_accuracy_threshold": 0.7745091915130615,
        "similarity_ap": 0.6484016684922852,
        "similarity_f1": 0.6789940828402367,
        "similarity_f1_threshold": 0.6167908906936646,
        "similarity_precision": 0.5510204081632653,
        "similarity_recall": 0.884393063583815
      }
    ]
  },
  "task_name": "FarsTail"
}
```

**Comment by memray** [comment]
Hi @KennethEnevoldsen ,

I was running the 'Ocnli' PairClassification task in MTEB_ZH, and it errors out due to:
```
mteb/abstasks/AbsTaskPairClassification.py", line 30, in _add_main_score
scores["main_score"] = scores[self.metadata.main_score]
KeyError: 'max_accuracy'
```
The content of the `scores` dict matches what you shared above. Apparently `max_accuracy` is not a valid metric name for this task. Is it a task-specific issue? Is there any official doc showing the main score metric of each task?

Thanks,
Rui

**Comment by Samoed** [comment]
@memray Sorry, that my changes cause problems. I've opened PR that fix this #1175

**Comment by memray** [comment]
@Samoed thanks for the prompt reply. I think `Cmnli` has the same issue. Now I hardcode the `self.metadata.main_score="max_ap"` as a work-around. 
BTW, should we use "max_ap" or "max_accuracy" as the main metric for Pair-Cls?

**Comment by Samoed** [comment]
Changes in new PR should fix everything

**Pull request merged**